e2e/test_operator_upgrades/operator_upgrades_test.go

```diff
@@ -386,6 +403,70 @@ var _ = Describe("Operator Upgrades", Label("e2e"), func() {
 		fixtures.GenerateUpgradeTableEntries(testOptions),
 	)
 
+	DescribeTable(
+		"upgrading a cluster where a storage and multiple stateless processes get restarted during the staging phase",
+		func(beforeVersion string, targetVersion string) {
+			isAtLeast := factory.OperatorIsAtLeast(
+				"v1.14.0",
+			)
+
+			if !isAtLeast {
+				Skip("operator doesn't support feature for test case")
+			}
+
+			if fixtures.VersionsAreProtocolCompatible(beforeVersion, targetVersion) {
+				Skip("this test case only affects version incompatible upgrades")
+			}
+
+			clusterSetup(beforeVersion, true)
+
+			// Select half of the stateless processes.
+			statelessPods := fdbCluster.GetStatelessPods()
+			Expect(statelessPods.Items).NotTo(BeEmpty())
+			selectedPods := fixtures.RandomPickPod(
+				statelessPods.Items,
+				len(statelessPods.Items)/2,
+			)
+
+			// Select a random storage process, and append it to "selectedPods".
+			storagePods := fdbCluster.GetStoragePods()
+			Expect(storagePods.Items).NotTo(BeEmpty())
+			selectedPods = append(selectedPods, fixtures.RandomPickOnePod(statelessPods.Items))
+
+			// Disable the feature that the operator restarts processes. This allows us to restart the processes
+			// selected above once all new binaries are present.
+			fdbCluster.SetKillProcesses(false)
+
+			// Start the upgrade.
+			Expect(fdbCluster.UpgradeCluster(targetVersion, false)).NotTo(HaveOccurred())
+
+			// Wait until all process groups are in the staging phase and the new binaries are available.
+			Eventually(func() bool {
+				return fdbCluster.AllProcessGroupsHaveCondition(fdbv1beta2.IncorrectCommandLine)
+			}).WithTimeout(10 * time.Minute).WithPolling(2 * time.Second).Should(BeTrue())
+
+			// Restart the processes selected above to pickup the new configuration and run with the
+			// newer version.
+			for _, selectedPod := range selectedPods {
```

Comment:
What is the intention of this test case? Right now I don't see any value in adding this test case:

1.) We set those processes to not be restarted
2.) We start the upgrade
3.) We check that the processes are getting the `IncorrectCommandLine` condition 
4.) We directly restart the affected processes

Shouldn't we do some kind of validation between 3. and 4.? Otherwise this is the same as doing a "normal" upgrade with additional steps.

If we have a theory/hypothesis we want to prove we should document that and make the according assertions. Otherwise I think we should remove this test case. What do you think?

Reply:
I was merely following what is specified in the radar, don't have a deeper understanding of what this test is supposed to accomplish. Let's delete this test then.